sklearn/preprocessing/_data.py

```diff
@@ -3428,7 +3428,8 @@ def inverse_transform(self, X):
             "yeo-johnson": self._yeo_johnson_inverse_transform,
         }[self.method]
         for i, lmbda in enumerate(self.lambdas_):
-            with np.errstate(invalid="ignore"):  # hide NaN warnings
+            # raise RuntimeWarning if return NaNs
```

Comment:
Do you think it'll be useful to show a more informative warning for this context?

Reply:
Yes, we can. 

The current message (from Numpy, I believe) is:  

> "invalid value encountered in power"

I was thinking of something like this:

> "Some values in the inverse-transformed data are NaN, which may be due to extreme skewness in the data. Consider addressing outliers before applying the transformation."

The code could be like this:

```python
for i, lmbda in enumerate(self.lambdas_):
    with np.errstate(invalid="ignore"):
        X[:, i] = inv_fun(X[:, i], lmbda)
        if np.isnan(X[:, i]).any():
            # raise UserWarning if return NaNs
            warnings.warn(
                """Some values in the inverse-transformed data are NaN, which may be due to
                extreme skewness in the data. Consider addressing outliers before applying
                the transformation.""",
                UserWarning
            )```

Reply:
Could we only check the `nan` values when the exception is raised to avoid a scan on the data? A try/except might be better then.

Reply:
Something like this?

```python
        for i, lmbda in enumerate(self.lambdas_):
            with np.errstate(invalid='warn'):
                with warnings.catch_warnings(record=True) as captured_warnings:
                    X[:, i] = inv_fun(X[:, i], lmbda)
            if captured_warnings and np.isnan(X[:, i]).any():
                    warnings.warn(
                        f"""Some values in column {i} of the inverse-transformed data are NaN. This may be due to 
                        extreme skewness or outliers in the data for this column. Consider addressing these issues, 
                        such as removing or imputing outliers, before applying the transformation.""",
                        UserWarning
                    )            
```
I believe the catch_warnings would fit better than the try-except because we want the results to be returned. I have also included the column number to help users identify where the issue is.

Reply:
```python
for i, lmda in enumerate(self.lambda_):
    with np.errstate(invalid="raise"):
        try:
            X[:, i] = inv_fun(X[:, i], lmbda)
        except Exception:
            if np.isnan(X[:, i]).any():
                warning.warn(...)
```

I think that we could still specialize the type of the `Exception` depending on what is raised. Also, is there any other "invalid" exception that would be raised or in other words, do we really need the check for the `nan` or can we just always raise the warning?